resource/Resource.md

```diff
@@ -0,0 +1,150 @@
+`Resource` MUST have getters for retrieving all the information used in `Resource` definition.
+
+Example in Go:
+```go
+type Resource {
+	Type   string
+	Labels map[string]string
+}
+```
+
+## Populating resources
+Resource information MAY be populated at any point between startup of the instrumented
+application and passing it to a backend-specific exporter. This explicitly includes
+the path through future OpenCensus components such as agents or services.
+
+For example, process-identifying information may be populated through the library while
+an agent attaches further labels about the underlying VM, the cluster, or geo-location.
+
+### From environment variables
+Population of resource information from environment variables MUST be provided by the
+core library. It provides the user with an ubiquitious way to manually provide information
+that may not be detectable automatically through available integration libraries.
+
+Two environment variables are used:
+* `OC_RESOURCE_TYPE`: defines the resource type. Leading and trailing whitespaces are trimmed.
+* `OC_RESOURCE_LABELS`: defines resource labels as a comma-seperated list of key/value pairs
+(`[ <key>="value" [ ,<key>="<value>" ... ] ]`). `"` characters in values MUST be escaped with `\`.
+
+For example:
+* `OC_RESOURCE_TYPE=k8s.io/container`
+* `OC_RESOURCE_LABELS=k8s.io/pod/name="pod-xyz-123",k8s.io/container/name="c1",k8s.io/namespace/name="default"`
+
+Population from environment variables MUST be the first applied detection process unless
+the user explicitly overwrites this behavior.
+
+### Auto-detection
+Auto-detection of resource information in specific environments, e.g. specific cloud
+vendors, MUST be implemented outside of the core libraries in third party or
+[census-ecosystem][census-ecosystem] repositories.
+
+### Merging
+As different mechanisms are run to gain information about a resource, their information
+has to be merged into a single resulting resource.
+Already set labels or type fields MUST NOT be overwritten. Label key namespaceing SHOULD
+be used to prevent collisions across different resource detection steps.
+
+### Detectors
+To make auto-detection implementations easy to use, the core resource package SHOULD define
+an interface to retrieve resource information. Additionally, helper functionality MAY be
+provided to effectively make use of this interface.
+The exact shape of those interfaces and helpers SHOULD be idiomatic to the respective language.
+
+Example in Go:
+
+```go
+type Detector func(context.Context) (*Resource, error)
+
+// Returns a detector that always returns a specific resource.
+func NewDetectorFromResource(*Resource) Detector
+
+// Returns a detector that runs all input detectors sequentially and merges their results.
+func ChainedDetector(...Detector) Detector
+```
+
+### Updates
+Over the runtime of an application, attributes of a resource may change in some cases.
```

Comment:
Here this sentence implies we support dynamic changes of the resource fields. Or maybe I am just wrong in understanding this.